catroid/src/org/catrobat/catroid/ui/ScriptActivity.java

```diff
@@ -259,9 +259,15 @@ public boolean onOptionsItemSelected(MenuItem item) {
 				break;
 
 			case R.id.unpacking:
-				Intent intent = new Intent(currentFragment.getActivity(), BackPackActivity.class);
-				intent.putExtra(BackPackActivity.EXTRA_FRAGMENT_POSITION, FRAGMENT_SOUNDS);
-				startActivity(intent);
+				if (currentFragment == lookFragment) {
```

Comment:
Do you really want to use `==` here instead of `instanceof` or `.equals()` (depending on what you want, since these do different things)? Just checking if this was on purpose.